### ISSUE TITLE
Offer Flow-based Client API

### ISSUE THREAD

**Issue by derolf** [akka/akka-grpc#550]
Right now, all the client-side API calls are of the form

```
serviceMethod[A, B](in: Source[A]) : Source[B]
```

However, to be able to process gRPC-calls inside of Graphs, we need a `Flow[A, B]`.

So, instead we can have:

```
serviceMethod: Flow[A, B]
```

**Comment by raboof** [comment]
I agree we should make this much easier. A helper/pattern as discussed in https://github.com/akka/akka/issues/26592 would already be useful, but this seems common enough that is it worth adding this to the generated gPRC API as well.

This leaves a number of questions to answer:
* what should the materialized value of this `Flow` be? In the related ticket #571 it was proposed to 'forward' the incoming materialized value.
* should we implement this as a client-specific method (either directly on the client class or on the 'lifted' API), or apply this change to both the server and the client API's?

This probably needs some experimentats to understand the impact on the user.

**Comment by viktorklang** [comment]
@raboof I'm not sure that the resulting materialized value matters, as it doesn't matter with the current implementation signature (It's always NotUsed).

**Comment by raboof** [comment]
For the server side I agree it is not interesting. For the client, I agree with your argument in #571 that it is weird to 'lose' the materialized value of the supplied `Source` ;).
